packages/cosmos-bin/src/cw3.rs

```diff
@@ -0,0 +1,434 @@
+struct AddMemberMessageOpt {
+    /// Members to add
+    #[clap(long)]
+    add: Vec<Address>,
+    /// Members to remove
+    #[clap(long)]
+    remove: Vec<Address>,
+    /// CW4 group contract address
+    #[clap(long)]
+    group: Address,
+}
+
+async fn update_members_message(
+    AddMemberMessageOpt { add, remove, group }: AddMemberMessageOpt,
+) -> Result<()> {
+    let msg = cw4_group::msg::ExecuteMsg::UpdateMembers {
+        add: add
+            .into_iter()
+            .map(|x| Member {
+                addr: x.get_address_string(),
+                weight: 1,
+            })
+            .collect(),
+        remove: remove.into_iter().map(|x| x.get_address_string()).collect(),
+    };
+    let msg = CosmosMsg::<Empty>::Wasm(WasmMsg::Execute {
+        contract_addr: group.get_address_string(),
+        msg: to_binary(&msg)?,
+        funds: vec![],
+    });
+    println!("{}", serde_json::to_string(&msg)?);
+    Ok(())
+}
+
+#[derive(clap::Parser)]
+struct ProposeOpt {
+    /// CW3 group contract address
+    #[clap(long)]
+    cw3: Address,
+    #[clap(flatten)]
+    tx_opt: TxOpt,
+    /// Title
+    #[clap(long)]
+    title: String,
+    /// Description, defaults to title
+    #[clap(long)]
+    description: Option<String>,
+    /// Messages, in JSON format
+    #[clap(long)]
+    msg: Vec<String>,
+}
+
+async fn propose(
+    cosmos: Cosmos,
+    ProposeOpt {
+        cw3,
+        tx_opt,
+        title,
+        description,
+        msg,
+    }: ProposeOpt,
+) -> Result<()> {
+    let wallet = tx_opt.get_wallet(cosmos.get_address_type());
+    let cw3 = cosmos.make_contract(cw3);
+    let res = cw3
+        .execute(
+            &wallet,
+            vec![],
+            cw3_flex_multisig::msg::ExecuteMsg::Propose {
+                description: description.unwrap_or_else(|| title.clone()),
+                title,
+                msgs: msg
+                    .into_iter()
+                    .map(|x| {
+                        serde_json::from_str::<CosmosMsg<Empty>>(&x)
+                            .context("Invalid CosmosMsg provided")
+                    })
+                    .collect::<Result<_>>()?,
+                latest: None,
+            },
+        )
+        .await?;
+    log::info!("Added in {}", res.txhash);
+    Ok(())
+}
+
+#[derive(clap::Parser)]
+struct ListOpt {
+    /// CW3 group contract address
+    #[clap(long)]
+    cw3: Address,
+}
+
+async fn list(cosmos: Cosmos, ListOpt { cw3 }: ListOpt) -> Result<()> {
+    let cw3 = cosmos.make_contract(cw3);
+    let mut start_after = None;
+    loop {
+        let ProposalListResponse::<Empty> { proposals } = cw3
+            .query(cw3_flex_multisig::msg::QueryMsg::ListProposals {
+                start_after,
+                limit: None,
+            })
+            .await?;
+        match proposals.last() {
+            None => break Ok(()),
+            Some(proposal) => start_after = Some(proposal.id),
+        }
+        for ProposalResponse {
```

Comment:
is there a benefit in destructuring in this case where most of the fields go unused?

Reply:
I personally like doing it so that it's explicit what I'm ignoring, but I'm pretty weird on that front. There's not really a tangible benefit. Also, using `{ foo, bar, .. }` syntax would be more idiomatic than my preferences.